watch/watcher.go

```diff
@@ -238,6 +243,7 @@ func (w *Watcher) observe() {
 		select {
 		case ev, ok := <-w.fsEvent:
 			if !ok {
+				w.done <- true
```

Comment:
just close(w.done) here. closed channels always return the zero value when received from